.github/workflows/benchmark-pull-request.yml

```diff
@@ -25,20 +25,41 @@ jobs:
           echo "USER_TAGS=pull_request_number:${{ github.event.issue.number }},repository:OpenSearch" >> $GITHUB_ENV
       - name: Check comment format
         id: check_comment
-        run: |
-          comment='${{ github.event.comment.body }}'
-          if echo "$comment" | jq -e 'has("run-benchmark-test")'; then
-            echo "Valid comment format detected, check if valid config id is provided"
-            config_id=$(echo $comment | jq -r '.["run-benchmark-test"]')
-            benchmark_configs=$(cat .github/benchmark-configs.json)
-            if echo $benchmark_configs | jq -e --arg id "$config_id" 'has($id)' && echo "$benchmark_configs" | jq -e --arg version "$OPENSEARCH_MAJOR_VERSION" --arg id "$config_id" '.[$id].supported_major_versions | index($version) != null' > /dev/null; then
-              echo $benchmark_configs | jq -r --arg id "$config_id" '.[$id]."cluster-benchmark-configs" | to_entries[] | "\(.key)=\(.value)"' >> $GITHUB_ENV
-            else
-              echo "invalid=true" >> $GITHUB_OUTPUT
-            fi
-          else
-            echo "invalid=true" >> $GITHUB_OUTPUT
-          fi
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const fs = require('fs');
+            const comment = context.payload.comment.body;
+            let commentJson;
+            try {
+              commentJson = JSON.parse(comment);
+            } catch (error) {
+              core.setOutput('invalid', 'true');
+              return;
+            }
+            if (!commentJson.hasOwnProperty('run-benchmark-test')) {
+              core.setOutput('invalid', 'true');
+              return;
+            }
+            const configId = commentJson['run-benchmark-test'];
+            let benchmarkConfigs;
+            try {
+              benchmarkConfigs = JSON.parse(fs.readFileSync('.github/benchmark-configs.json', 'utf8'));
```

Comment:
@rishabh6788 I have a concern here, may be you could help me to clear it out: change for `pull_request_target` is fine (I think this is what really need), but would it apply to non-workflow files? In other words,   `.github/benchmark-configs.json` is coming from pull request or base branch?

Reply:
In respect to this workflow the file comes from base branch, not head as `issue_comment` runs in context of base branch only. See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issue_comment,  check `GITHUB_SHA` and `GITHUB_REF` fields.